src/main/java/frc/robot/poseestimator/limelight/MultiLimelight.java

```diff
@@ -0,0 +1,111 @@
+package frc.robot.poseestimator.limelight;
+
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import frc.robot.Robot;
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class MultiLimelight extends GBSubsystem {
+
+    private static MultiLimelight instance;
+
+    private List<Limelight> limelights;
+    private Pose2d currentPosition;
+
+    private MultiLimelight() {
+        super("MultiLimeLight/");
+        limelights = new ArrayList<>();
+        for (String limelightName : VisionConstants.LIMELIGHT_NAMES) {
+            limelights.add(new Limelight(limelightName));
+        }
+    }
+
+    public static void init() {
+        instance = new MultiLimelight();
+    }
+
+    public static MultiLimelight getInstance() {
+        if (instance == null) {
+            init();
+        }
+        return instance;
+    }
+
+    public List<Optional<Pair<Pose2d, Double>>> getAll2DEstimates() {
+        ArrayList<Optional<Pair<Pose2d, Double>>> estimates = new ArrayList<>();
+
+        for (Limelight limelight : limelights) {
+            currentPosition = Robot.getPosEstimator.getOdometryPose();
+            if (limelight.hasTarget()) {
+                if (getLimelightedOutputInTolerance(limelight)) {
+                    estimates.add(limelight.getUpdatedPose2DEstimation());
+                }
+            }
+        }
+
+        return estimates;
+    }
+
+    public boolean getLimelightedOutputInTolerance(Limelight limelight) {
+        Pose2d limelightPosition;
+        Transform2d transformDifference;
+        Rotation2d rotationDifference;
+
+        if (limelight.getAprilTagConfidence())
+            if (limelight.getUpdatedPose2DEstimation().isPresent()) {
+                limelightPosition = limelight.getUpdatedPose2DEstimation().get().getFirst();
+                transformDifference = limelightPosition.minus(currentPosition);
+                rotationDifference = limelightPosition.getRotation().minus(currentPosition.getRotation());
+
+                return transformDifference.getTranslation().getNorm() >= VisionConstants.POSITION_TOLERANCE
+                        && rotationDifference.getDegrees() >= VisionConstants.ROTATION_TOLERANCE.getDegrees();
+            }
+
+        return false;
+    }
```

Comment:
split into many smaller methods, one for the confidence of the output, one for the difference tolerance, and so on

Reply:
I have redistribute the if statements, is it ok now?

Reply:
yes 👍